libtmux/test.py

```diff
@@ -43,6 +46,52 @@ def retry(seconds=RETRY_TIMEOUT_SECONDS):
     return (lambda: time.time() < time.time() + seconds)()
 
 
+def retry_until(
+        fun,
+        seconds=RETRY_TIMEOUT_SECONDS,
+        *,
+        interval=RETRY_INTERVAL_SECONDS,
+        raises=True,
+):
+    """
+    Retry a function until a condition meets or the specified time passes.
+
+    Parameters
+    ----------
+    fun : callable
+        A function that will be called repeatedly until it returns ``True``  or
+        the specified time passes.
+    seconds : int
```

Comment:
```suggestion
    seconds : float
```

Based on [`time.sleep()`](https://github.com/python/typeshed/blob/740193a/stdlib/time.pyi#L76) annotation